tests/Tokenizer/TokensTest.php

```diff
@@ -869,6 +869,85 @@ public function assertFindBlockEnd($expectedIndex, $source, $type, $searchIndex)
     }
 
     /**
+     * @param string $expected   valid PHP code
+     * @param string $input      valid PHP code
+     * @param int    $index      token index
+     * @param int    $offset
+     * @param string $whiteSpace white space
+     *
+     * @dataProvider provideEnsureWhitespaceAtIndexCases
+     */
+    public function testEnsureWhitespaceAtIndex($expected, $input, $index, $offset, $whiteSpace)
+    {
+        $tokens = Tokens::fromCode($input);
+        $tokens->ensureWhitespaceAtIndex($index, $offset, $whiteSpace);
+
+        $this->assertSame($expected, $tokens->generateCode());
```

Comment:
I think it's not working correctly.
Could you please take over the https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/2771
and fix 2.2 line as well if needed ?